src/Client.ts

```diff
@@ -177,54 +181,172 @@ export class Client extends ClientJS {
+    // # init global commands
+    const existing = (await this.fetchSlash()).filter((s) => !s.guild);
+    const slashes = this.slashes.filter((s) => !s.guilds?.length);
+
+    const added = slashes.filter(
+      (s) => !existing.find((c) => c.name === s.name)
+    );
+    const updated = slashes
+      .map<[ApplicationCommand, DSlash]>((s) => [
+        existing.find((c) => c.name === s.name),
+        s,
+      ])
+      .filter((s) => s[0]);
+    const deleted = existing.filter((c) =>
+      slashes.every((s) => s.name !== c.name)
     );
+
+    if (!this.silent)
+      console.log(
+        `${this.user.username} >> global >> command >> Adding ${
+          added.length
+        } [${added.map((s) => s.name).join(", ")}]`
+      );
+    if (!this.silent)
+      console.log(
+        `${this.user.username} >> global >> command >> deleting ${
+          deleted.size
+        } [${deleted.map((s) => s.name).join(", ")}]`
+      );
+    if (!this.silent)
+      console.log(
+        `${this.user.username} >> global >> command >> updating ${updated.length}`
+      );
+
+    await Promise.all([
+      ...added.map((s) => this.application.commands.create(s.toObject())),
+      ...updated.map((s) => s[0].edit(s[1].toObject())),
+      ...deleted.map((key) => this.application.commands.delete(key)),
+    ]);
+
+    //    await Promise.all(
+    //      this.slashes.map(async (slash) => {
+    //        // Init all the @Slash
+    //        if (slash.guilds.length > 0) {
+    //          // If the @Slash is guild specific, add it to the guild
+    //          await Promise.all(
+    //            slash.guilds.map(async (guildID) => {
+    //              const guild = this.guilds.cache.get(guildID as Snowflake);
+    //
+    //              if (!guild) {
+    //                throw new GuildNotFoundError(guildID);
+    //              }
+    //
+    //              const commands = guild.commands;
+    //              const command = await commands.create(slash.toObject());
+    //
+    //              if (slash.permissions.length <= 0) return;
+    //
+    //              // https://discord.js.org/#/docs/main/master/class/ApplicationCommandPermissionsManager?scrollTo=set
+    //              await commands.permissions.set({
+    //                command: command,
+    //                permissions: slash.getPermissions(),
+    //              });
+    //            })
+    //          );
+    //        } else {
+    //          // If the @Slash is global, add it globaly
+    //          const commands = this.application.commands;
+    //          await commands.create(slash.toObject());
+    //
+    //          // Only available for Guilds
+    //          // https://discord.js.org/#/docs/main/master/class/ApplicationCommand?scrollTo=setPermissions
+    //          // if (slash.permissions.length <= 0) return;
+    //
+    //          // await commands.setPermissions(command, slash.getPermissions());
+    //        }
+    //      })
+    //    );
```

Comment:
Please remove this, we'll have it in commit history if need be.

Reply:
okay